src/Icons/src/IconRenderer.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Icons;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @internal
+ */
+final class IconRenderer
+{
+    public function __construct(
+        private IconRegistryInterface $registry,
+        private array $defaultIconAttributes = [],
+    ) {
+    }
+
+    /**
+     * @param array<string,string|bool> $attributes
+     */
+    public function renderIcon(string $name, array $attributes = []): string
+    {
+        [$content, $iconAttr] = $this->registry->get($name);
+
+        $iconAttr = array_merge($iconAttr, $this->defaultIconAttributes);
+
+        return sprintf(
+            '<svg%s>%s</svg>',
+            self::normalizeAttributes([...$iconAttr, ...$attributes]),
+            $content,
+        );
+    }
+
+    /**
+     * @param array<string,string|bool> $attributes
+     */
+    private static function normalizeAttributes(array $attributes): string
```

Comment:
We'll need some html escape there

Minor: i'd rename this renderAttributes() as it returns HTML

Reply:
Can you give an example so I can make a test

Reply:
If the user calls `$renderer->renderIcon('star', ['data-foo' => 'a "string" with a quote'])`, the end-result should be escaped for the HTML attributes, since the return value from this function is meant to be rendered raw. Apparently this is how we do it in StimulusBundle https://github.com/symfony/stimulus-bundle/blob/2.x/src/Dto/StimulusAttributes.php#L123

Reply:
Oh man, this strategy escapes valid attribute values (like spaces):

```diff
-viewBox="0 0 24 24" 
+viewBox="0&#x20;0&#x20;24&#x20;24"
```

Is this what we want?

Reply:
If you use Escaper, you need to use the "html_attr" strategy

Reply:
I copied the code from the stimulus bundle (uses `html_attr`) and that was the result.

Reply:
I used a more direct method in my Icon object PR